src/main/java/apoc/nodes/Nodes.java

```diff
@@ -136,6 +137,134 @@ public boolean connected(@Name("start") Node start, @Name("start") Node end, @Na
         }
     }
 
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.nodes.collapse([nodes...],[{properties:'overwrite' or 'discard' or 'combine'}]) yield from, rel, to merge nodes onto first in list")
+    public Stream<VirtualPathResult> collapse(@Name("nodes") List<Node> nodes, @Name(value = "config", defaultValue = "") Map<String, Object> config) {
+        if (nodes == null || nodes.isEmpty() || nodes.size() < 2) return Stream.empty();
+        RefactorConfig conf = new RefactorConfig(config);
+        final Node firstNode = nodes.get(0);
+        Set<Node> nodeSet = new HashSet<>(nodes);
+        final VirtualNode first = createVirtualRelationshipFirstNode(mergedVirtualNode(nodeSet, conf), firstNode);
+        nodeSet.stream().skip(1).distinct().forEach(node -> mergeVirtualNodes(node, first, firstNode, conf));
+        return StreamSupport.stream(first.getRelationships().spliterator(), false)
+                .map(relationship -> {
+                    Node start = nodeSet.contains(relationship.getStartNode()) ? first : relationship.getStartNode();
+                    Node end = nodeSet.contains(relationship.getEndNode()) ? first : relationship.getEndNode();
+                    return new VirtualPathResult(start, relationship, end);
+                });
+    }
+
+    private VirtualNode createVirtualRelationshipFirstNode(VirtualNode first, Node firstNode) {
+        firstNode.getRelationships(Direction.OUTGOING).forEach(relationship ->{
+            VirtualRelationship virtualRelationship = first.createRelationshipTo(relationship.getEndNode(), relationship.getType());
+            for (Map.Entry<String, Object> prop : relationship.getAllProperties().entrySet())
+                virtualRelationship.setProperty(prop.getKey(), prop.getValue());
+        });
+
+        firstNode.getRelationships(Direction.INCOMING).forEach(relationship -> {
+            VirtualRelationship virtualRelationship = first.createRelationshipFrom(relationship.getStartNode(), relationship.getType());
+            for (Map.Entry<String, Object> prop : relationship.getAllProperties().entrySet())
+                virtualRelationship.setProperty(prop.getKey(), prop.getValue());
+        });
+
+        return first;
+    }
+
+    private static VirtualNode mergedVirtualNode(Set<Node> nodes, RefactorConfig conf) {
+        Create create = new Create();
+        VirtualNode virtualNode = (VirtualNode) create.vNodeFunction(null, Util.map());
```

Comment:
Passare le label e propertiestdirettamente nell argomento:
```suggestion
        List<String> labels = <lista>;
        Map<String, Object> props = <props>;
        VirtualNode virtualNode = (VirtualNode) create.vNodeFunction(labels, props);
```
Dare sempre uno sguardo al metodo che si usa...